### ISSUE TITLE
Learning rate scheduler confusion

### ISSUE THREAD

**Issue by opcode81** [aai-institute/tianshou#1197]
The way in which learning rate schedulers are parametrised has led to confusion.
Particularly torch's `LambdaLR` is used throughout Tianshou. While we use it correctly (as far as I can tell), the terminology that is used can confuse users (and has confused a user in #1157). The reason for the confusion is that the learning rate scheduler has a different notion of "epoch" than Tianshou itself does: 

For the learning rate scheduler, if its `step` method is called without the explicit `epoch` argument (which we do), the "epoch" is simply a counter of the number of times the `step` function has been called in the past. Since we call the `step` function once in every training step, the maximum "epoch" for the scheduler is the total number of training steps (which can be computed as `ceil(step_per_epoch / step_per_collect) * num_epochs`). We thus correctly parametrise the scheduler as follows (e.g. in atari_ppo.py):

https://github.com/thu-ml/tianshou/blob/bd74273f3348932eb0875a6c9dc50d8646d494c5/examples/atari/atari_ppo.py#L128-L132

A user thinking it is the Tianshou "epoch" semantics (like the reporter of #1157), will be tempted to do this instead:

```python
lr_scheduler = LambdaLR(optim, lr_lambda=lambda epoch: 1 - epoch / args.epoch)
```

The fact that we use the term `epoch` in our code for parametrising schedulers is the main issue here.

Note that if a user defines the scheduler as above, because the number of training steps is much larger than the number of epochs, the lambda factor becomes negative. This is not caught in `LambdaLR` and effectively causes the gradient to be reversed, resulting in fatal learning failure.

I would suggest to:
* Replace the term `epoch` with the term `update_num` or `training_step` (also in the high-level API)
* Try to catch errors where the learning rate becomes negative.
* Improve documentation (where appropriate) in order to avoid this kind of confusion

**Comment by MischaPanch** [comment]
Loosely related: #1198

**Comment by opcode81** [comment]
This problem has been eliminated with the introduction of `OptimizerFactory` and `LRSchedulerFactory` in v2.
